database/USAFacts_collector

```diff
@@ -0,0 +1,105 @@
+covid_usa_deaths$`County Name` <- gsub(" County", "", covid_usa_cases$`County Name`, fixed = T)
+
+### Change Alaska municipality, borough, etc names to match lat/long dataset
+AK_current <- c("Municipality of Anchorage" , "Fairbanks North Star Borough", "City and Borough of Juneau",  
+                "Kenai Peninsula Borough" , "Ketchikan Gateway Borough" , "Matanuska-Susitna Borough")
+
+AK_corrected <- c("Anchorage", "Fairbanks North Star", "Juneau", "Kenai Peninsula", "Ketchikan Gateway", "Matanuska-Susitna")
+
+for (i in 1:length(AK_current)){
+  hit <- which(covid_usa_cases$`County Name` == AK_current[i])
+  covid_usa_cases$`County Name`[hit] <- AK_corrected[i]
+}
+
+### Change state names to abbreviations in latlong dataframe
+county_latlong$Province_State <- state.abb[match(county_latlong$Province_State, state.name)]
+
+### Separate case/death counts data for easier loop coding below
+cases <- covid_usa_cases[5:ncol(covid_usa_cases)]
+deaths <- covid_usa_deaths[5:ncol(covid_usa_cases)]
+
+#### Create empty NECSI-formatted datasets to populate with transformed data ####
+### One empty data frame serves as final master sheet
+### Other data frame is populated with information for each county which is then appended to the master sheet
+covid19_usa_NECSI <- data.frame(matrix(ncol = 14, nrow = 0))
+colnames(covid19_usa_NECSI) <- c("date", "county", "county_FIPS", "state", "country", "lat/long", "confirmed", "recovered", 
+                                "dead", "daily_diff_confirm", "daily_diff_recover", "daily_diff_dead", "admin_level", 
+                                "source")
+county_NECSI <- covid19_usa_NECSI
+
+#### Loops for populating NECSI time series-formatted data frame ####
+### Outer loop cycles through counties, inner loop populates relevant information for each county for each date available
+### in the USAFacts data sheet
+for (k in 1:length(covid_usa_cases$`County Name`)){
+
+for (i in 1:ncol(cases)){
+  day <- colnames(cases)[i]
+  county <- as.character(covid_usa_cases[k, 2])
+  countyFIPS <- as.character(covid_usa_cases[k, 1])
+  state <- as.character(covid_usa_cases[k, 3])
+  
+county_NECSI[i, 1] <- day
+county_NECSI[i, 2] <- county
+county_NECSI[i, 3] <- countyFIPS
+county_NECSI[i, 4] <- state
+
+# Find matching row with relevant county lat/lon information if county is not "Statewide Unallocated" and associate 
+# relevant lat/long coordinates
+if (county == "Statewide Unallocated"){
+  county_NECSI[i, 6] <- NA # Can be replaced with state-wide lat/long?
+} else {
+county_match <- which(county_latlong$Admin2 == county & county_latlong$Province_State == state)
+county_NECSI[i, 6] <- paste(county_latlong$Lat[county_match], ",", county_latlong$Long_[county_match], sep = "")
+}
+
+county_NECSI[i, 7] <- cases[k,i]
+# No data in USAFacts for number of recovered, cell [i, 10] skipped
+county_NECSI[i, 9] <- deaths[k, i]
+if (day == "1/22/20"){
```

Comment:
you should document "magic" parameters -- why "1/22/20"?
Even better, just write code to take the first day.

Reply:
Ok, will fix